### PR TITLE
feat/added async support for isAllowed prop for createStripeGraphQLServer

### PR DESCRIPTION
**Pull request by ejkkan**
fixes issue: https://github.com/nhost/nhost/issues/980

### CODE REVIEW

**Review by plmercereau — CHANGES_REQUESTED** [review]
Thanks for this!
You also need to add an `await` when calling this function [here](https://github.com/nhost/nhost/blob/66ef8f91a3c3430ec877a2d7cbaf2f64df90338b/packages/stripe-graphql-js/src/schema/stripe.ts#L20), [here](https://github.com/nhost/nhost/blob/66ef8f91a3c3430ec877a2d7cbaf2f64df90338b/packages/stripe-graphql-js/src/schema/stripe.ts#L67), and [there](https://github.com/nhost/nhost/blob/66ef8f91a3c3430ec877a2d7cbaf2f64df90338b/packages/stripe-graphql-js/src/schema/stripe.ts#L98)

**Comment by ejkkan** [comment]
> Thanks for this! You also need to add an `await` when calling this function [here](https://github.com/nhost/nhost/blob/66ef8f91a3c3430ec877a2d7cbaf2f64df90338b/packages/stripe-graphql-js/src/schema/stripe.ts#L20), [here](https://github.com/nhost/nhost/blob/66ef8f91a3c3430ec877a2d7cbaf2f64df90338b/packages/stripe-graphql-js/src/schema/stripe.ts#L67), and [there](https://github.com/nhost/nhost/blob/66ef8f91a3c3430ec877a2d7cbaf2f64df90338b/packages/stripe-graphql-js/src/schema/stripe.ts#L98)

Thanks. Please have another look when you have the time @plmercereau

File: packages/stripe-graphql-js/src/schema/stripe.ts [hunk 1]
```diff
@@ -64,7 +64,7 @@ builder.objectType('Stripe', {
           starting_after: startingAfter || undefined
         })
 
-        const customerData = customers.data.filter((customer) => isAllowed(customer.id, context))
+        const customerData = customers.data.filter(async (customer) => await isAllowed(customer.id, context))
```
Comment (plmercereau) [inline packages/stripe-graphql-js/src/schema/stripe.ts:67]: Unfortunately this won't work as `filter` doesn't support async functions :(
We could solve this with a reduced and `Promise.all`, but for the sake of readability, we could do something like this:

```suggestion
        const customerData: Stripe.Customer[] = []
        for (const customer of customers.data) {
          if (await isAllowed(customer.id, context)) {
            customerData.push(customer)
          }
        }
```

**Review by plmercereau — CHANGES_REQUESTED** [review]

File: packages/stripe-graphql-js/src/schema/stripe.ts (hunk 1 above)
Comment (ejkkan) [reply packages/stripe-graphql-js/src/schema/stripe.ts:67]: My bad. Thanks for the suggestion.
Lets go for it :)

**Comment by plmercereau** [comment]
Looks good to me! I added a changeset so a new version is released.
@elitan if you don't mind having a look before I merge the PR

**Review by plmercereau — APPROVED** [review]

**Review by elitan — APPROVED** [review]
LGTM

**Pull request merged**